src/Notification/ToastNotification.svelte

```diff
@@ -66,14 +71,15 @@
   }
 
   onMount(() => {
-    if (timeout) {
-      timeoutId = setTimeout(() => close(true), timeout);
-    }
-
     return () => {
       clearTimeout(timeoutId);
     };
   });
+
+  $: if (timeout) {
+    clearTimeout(timeoutId);
+    timeoutId = setTimeout(() => close(true), timeout);
+  }
```

Comment:
- This would also restart the timeout while the toast is closed and then potentially re-fire the event.
- Does not clear an existing timeout when set to `0` because the `if` would be `false`. Maybe just remove the condition.
- Probably requires something like a `mounted` flag guard (set in `onMount`) to prevent running in SSR.

Reply:
- Could you give an example?
- The default `timeout` prop value is `0`, so the `if` statement here is intentional (`0` being falsy)
- Good flag on SSR, however, an SSR guard isn't required here as setTimeout/clearTimeout are supported in Node.js

Reply:
- Just any regular set on the `timeout` after the toast has been closed, would trigger the reactive statement which schedules a timeout which in turn causes the event to fire. This is currently unlikely to happen intentionally because there is no way to reopen the component, as `open` is not accessible.
  In general the `open` state should probably be checked before scheduling the timeout.
- I know why the if is there, but it should only guard the `setTimeout`, not the `clearTimeout`.
  Right now, once the timeout has started, setting it to `0` will *not* cause `clearTimeout` to run, so it's impossible to properly stop the automatic close logic once it has started.
- Regarding SSR, one should not let timeouts like this start on the server, they serve no purpose can only cause problems (e.g. wasting resources, causing errors and taking the server with them, keeping things alive that should not be), right now the timeout will not be cleared properly either.

By the way, I would not be opposed to just dropping this automatic close logic from the component entirely.

If `open` were to be exposed, it would be possible to do this in user code and having to deal with this reactively here (because properties *might* change) just complicates things. It is already possible to do this user code to some degree by just destroying the component after a timeout (when using the client-side component API).

Reply:
I've amended the code to address the following:

- Only start the timer if `open` is not already `false`
- Omit the `clearTimeout`
- Add in an SSR guard as we don't need this to run server-side